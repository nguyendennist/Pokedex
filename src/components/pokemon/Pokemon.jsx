import React, { Component } from 'react';
import PropTypes from 'prop-types';
import PokemonAbilities from 'components/pokemon-abilities/PokemonAbilities.jsx';

export default class Pokemon extends Component {
	constructor(props) {
		super(props);

		this.state = {
			showMoreInfo: 'none'
		};

		this.onClick = this.onClick.bind(this);
	}

	onClick() {
		this.setState(prevState => {
			const newState = {};

			if (prevState.showMoreInfo === 'none') {
				newState.showMoreInfo = 'inline-block';
			} else {
				newState.showMoreInfo = 'none';
			}

			return newState;
		});
	}

	render() {
		const style = {
			display: this.state.showMoreInfo
		};

		return (
			<div className='pokemon' onClick={this.onClick}>
				<div
					className='pokemon-height-and-weight'
					style={style}
				>
					<div>
						<p>Height<br/>{this.props.pokemonData.height}</p>
						<p>Weight<br/>{this.props.pokemonData.weight}</p>
					</div>
				</div>
				<div className='pokemon-image-and-name'>
					<img
						alt={`A pokemon named ${this.props.pokemonData.name}`}
						src={this.props.pokemonData.sprites.front_default}
					/>
					<p className='pokemon-text'>{this.props.pokemonData.name}</p>
				</div>
				<PokemonAbilities
					abilities={this.props.pokemonData.abilities}
					style={style}
				/>
			</div>
		);
	}
}

Pokemon.propTypes = {
	pokemonData: PropTypes.object.isRequired
};
