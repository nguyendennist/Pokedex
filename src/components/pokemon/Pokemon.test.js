import React from 'react';
import { create } from 'react-test-renderer';
import Pokemon from './Pokemon.jsx';

describe('unit tests', () => {
	test('should be defined', () => {
		expect(Pokemon).toBeDefined();
	});
});

describe('ui tests', () => {
	test('should render correctly', () => {
		const tree = create(
			<Pokemon key={'1'} pokemonData={{
				abilities: [
					{
						ability: {
							name: 'chlorophyll',
							url: 'https://www.pokeapi.co/api/v2/ability/34/'
						},
						is_hidden: true,
						slot: 3
					},
					{
						ability: {
							name: 'overgrow',
							url: 'https://www.pokeapi.co/api/v2/ability/65/'
						},
						is_hidden: false,
						slot: 1
					}
				],
				height: 7,
				id: 1,
				name: 'bulbasaur',
				sprites: {
					front_default: 'https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/1.png'
				},
				weight: 69,
			}} />
		).toJSON();

		expect(tree).toMatchSnapshot();
	});
});
