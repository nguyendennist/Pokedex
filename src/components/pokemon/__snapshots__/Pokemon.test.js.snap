// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ui tests should render correctly 1`] = `
<div
  className="pokemon"
  onClick={[Function]}
>
  <div
    className="pokemon-height-and-weight"
    style={
      Object {
        "display": "none",
      }
    }
  >
    <div>
      <p>
        Height
        <br />
        7
      </p>
      <p>
        Weight
        <br />
        69
      </p>
    </div>
  </div>
  <div
    className="pokemon-image-and-name"
  >
    <img
      alt="A pokemon named bulbasaur"
      src="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/1.png"
    />
    <p
      className="pokemon-text"
    >
      bulbasaur
    </p>
  </div>
  <div
    className="pokemon-abilities"
    style={
      Object {
        "display": "none",
      }
    }
  >
    <div>
      <p>
        Abilities
      </p>
      <p>
        chlorophyll
      </p>
      <p>
        overgrow
      </p>
    </div>
  </div>
</div>
`;
